src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -233,7 +233,14 @@ public string GetNodeLocation()
 
             string nodeFolder = NodeHandler.nodeFolder;
 
-            if (taskHasNode20Data)
+            if (taskHasNode20Data && !isNode20SupportedSystems())
+            {
+                Trace.Warning($"The operating system the agent is running on doesn't support Node20. " +
+                             "Please upgrade the operating system of this host to ensure compatibility with Node20 tasks: " +
```

Comment:
should be localized (put in resource file) -- also please post a screenshot of the message (or at least the text) to the internal channel and tag @geekzter for review

I think the message should mention that we're using the node16 runner instead